internal/storage/bucket_handle.go

```diff
@@ -210,11 +210,16 @@ func (b *bucketHandle) ListObjects(ctx context.Context, req *gcs.ListObjectsRequ
 	itr := b.bucket.Objects(ctx, query) // Returning iterator to the list of objects.
 	pi := itr.PageInfo()
 	pi.MaxSize = req.MaxResults
+	pi.Token = req.ContinuationToken
 	var list gcs.Listing
 
 	// Iterating through all the objects in the bucket and one by one adding them to the list.
 	for {
 		var attrs *storage.ObjectAttrs
+		if len(list.Objects) == req.MaxResults {
+			err = nil
```

Comment:
1. add a comment why this if condition is required
2. err is nil by default. Do we still need to set err=nil?

Reply:
fixed